internal/github/github.go

```diff
@@ -81,19 +79,31 @@ func (s *sleeper) sleep(d time.Duration) {
 type GHClient struct {
 	client  *github.Client
 	sleeper *sleeper
+	pendingRecheckTime time.Duration
 }
 
-func NewGithubClient(ctx context.Context, authInfo AuthInfo) (GHClient, error) {
+func NewGithubClient(ctx context.Context, authInfo AuthInfo, pendingRecheckTime time.Duration) (GHClient, error) {
+	var (
+		githubClient GHClient
+		err          error
+	)
+
 	// If a GitHub token is provided, use it to authenticate in preference to
 	// App authentication
 	if authInfo.GithubToken != "" {
 		log.Debug("Using GitHub token for authentication")
-		return AuthenticateWithToken(ctx, authInfo.GithubToken), nil
+		githubClient = AuthenticateWithToken(ctx, authInfo.GithubToken)
+	} else {
+		// Otherwise, use the App authentication flow
+		log.Debug("Using GitHub App for authentication")
+		githubClient, err = AuthenticateWithApp(ctx, authInfo.PrivateKey, authInfo.AppID, authInfo.InstallationID)
+		if err != nil {
+			return GHClient{}, err
+		}
```

Comment:
It's a matter of taste/style preference, but I'd prefer passing `pendingRecheckTime` down to `AuthenticateWithToken` / `AuthenticateWithApp`, and then we can avoid the `else`.

Reply:
Done.